### PR TITLE
Xcode additional features

### PR DESCRIPTION
**Pull request by andrewwillmott**
This pull request adds the following features to the Xcode generator:

- Adds support for 'debugcmd' if XcodeScheme is enabled

- option XcodeSchemeAppsOnly -- only adds schemes for projects that build an app. If you have a large solution with lots of internal libraries (like many games), this keeps the schemes menu under control. If there are any libraries that you do want to build separately, you can always add XcodeScheme just to them.

- option XcodeSchemePerConfig -- Adds a separate scheme for each project configuration. This allows quick switching between the config to build without having to edit the scheme.

- option XcodeGroupLocations -- sets the group-relative location properly for both files and folders/groups.  This makes  projects easier to navigate (especially if temporarily adding new files via IDE),  and slightly reduces project size on big projects.

- Updates to the docs for the above, and also to address breakage on non-github-Markdown formatters. (Mostly adding new lines in front of lists where they didn't already exist, and avoiding using '_' with tokens that already contain a '_'.)

Personally, I would like to see at least XcodeScheme and XcodeGroupLocations become defaults, as I always use the debug* commands. (Plus it makes bgfx run out of the box on Xcode, rather than having to manually set a custom directory!) And, depending on other people's use cases, XcodeSchemePerConfig and maybe XcodeSchemeAppsOnly too, but that's a wider discussion. For now, with no options set, these changes do not change behaviour.

To see this options in operation on bgfx, you want to add this to scripts/genie.lua:

        language "C++"
        startproject "example-00-helloworld"
 
+       options { "XcodeSchemeAppsOnly", "XcodeSchemePerConfig", "XcodeGroupLocations" }
+
 MODULE_DIR = path.getabsolute("../")

Your schemes menu should then look something like this:

image

### CODE REVIEW

**Comment by bkaradzic** [comment]
cc @rhoot

File: src/actions/xcode/xcode10.lua [hunk 1]
```diff
@@ -98,7 +98,17 @@
...
+			if xcode.project_has_scheme(prj) then
+				if prj.options and prj.options.XcodeSchemePerConfig then
+					for cfg in premake.eachconfig(prj) do
+						premake.generate(prj, "%%.xcodeproj/xcshareddata/xcschemes/%% " .. cfg.name .. ".xcscheme",
+							function(prj) xcode.project_scheme(prj, cfg) end)
+					end
+				else
+					premake.generate(prj, "%%.xcodeproj/xcshareddata/xcschemes/%%.xcscheme", xcode.project_scheme)
+				end
+			end
```
Comment (rhoot) [inline src/actions/xcode/xcode10.lua]: I would prefer if you did this work in a function in xcode_scheme and just call that function here, so all this logic is not copy+pasted into each action and have to be kept up-to-date everywhere.

Comment (andrewwillmott) [reply src/actions/xcode/xcode10.lua]: I did it that way to keep the convention that the path generation calls are in the main action files, but sure, I've just pushed that change.

**Comment by rhoot** [comment]
:+1:

**Pull request merged**